mysticeti-core/src/net_sync.rs

```diff
@@ -187,7 +193,7 @@ impl<H: BlockHandler + 'static, C: CommitObserver + 'static> NetworkSyncer<H, C>
                 }
                 NetworkMessage::Block(block) => {
                     tracing::debug!("Received {} from {}", block.reference(), peer);
-                    if let Err(e) = block.verify(&inner.committee) {
+                    if let Err(e) = block.verify(&inner.committee, &statement_block_validator) {
```

Comment:
I would separate Mysticeti block verification and custom verification in separate if blocks - I think we can print different log messages so that we know which verification step failed